### ISSUE TITLE
Stuck at phase WaitingForKubeadmInit

### ISSUE THREAD

**Issue by nguyenthai0107** [kubernetes-sigs/cluster-api-provider-vsphere#2137]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
- cluster-api-providẻ-vsphere version v1.3.5
- Hello team 
I test init a cluster-api with 
Control Plane master: 1
Worker node : 2 

Everything ok ( It can create VM on Vsphere, and i can ssh into this Master ) 
But im get stuck at phase _WaitingForKubeadmInit_ when i run cmd
clusterctl describe cluster capi-quickstart 

![image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/assets/69754985/ca5627b0-e908-4be6-9fec-edff090acba8)

And i checked something , here is some collected log which was so strange
1. logs of capi-kubeadm-control-plane-controller-manager in namespace capi-kubeadm-control-plane-system with error X509

![image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/assets/69754985/c11411c1-b09b-4c21-97ed-f9139ab4f958)
With BLUE is a IP internal of K3S ( Cluster Management )
![image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/assets/69754985/011aab19-3ead-4f03-8e9a-dd8f0199a702)

2. i also can ssh into Master of workload Cluster, and get inside path 
/etc/kubernetes/admin.conf , take context and bring to another host -> .kube/config 
And when i kubectl get node or pod and also received x509 error too

**What did you expect to happen:**
Bypass phase  WaitingForKubeadmInit and Cluster status will fine everything

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: v1.3.5
- Kubernetes version: (use `kubectl version`): 1.22.12
- OS (e.g. from `/etc/os-release`):

**Comment by chrischdi** [comment]
I see two possible reasons for that:
* Either the VM's bootstrapping does not go through
* Or whatever does load balance the kube-apiserver for the control plane node is wrong. (e.g. kube-vip if used?!)

Suggestions:
* Could you take a look at the created ControlPlane VM (via ssh) and check if the bootstrapping went through? (e.g. check if kubelet is running (`systemctl status kubelet`), or if all static pod manifests are up and running (compare `/etc/kubernetes/manifest` to `crictl ps`).
* Could you provide information aobut the censored IP address? What does this direct to? Is this a load balancer?
  * Could you check whatever that is if its properly routing to **only** the new vm's?

**Comment by nguyenthai0107** [comment]
Hello @chrischdi 
Sorry for a long longggg time for reply.
Finally i can make it TRUE :)) 
![image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/assets/69754985/a84b4411-cd24-4a43-9a1f-fcacd07f5011)

**Comment by chrischdi** [comment]
So looks like you unblocked yourself?

Could you provide the info what did help you to resolve your problem?

**Comment by dhqanh-hpe** [comment]
> Hello [@chrischdi](https://github.com/chrischdi) Sorry for a long longggg time for reply. Finally i can make it TRUE :)) ![image](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/assets/69754985/a84b4411-cd24-4a43-9a1f-fcacd07f5011)

Hi @nguyenthai0107 , I'm facing your problem. Could you tell me that how to fix it?

**Comment by chrischdi** [comment]
@dhqanh-hpe : Not really getting it. From the screenshot all looks fine.
Are you hitting an issue? if yes maybe you should consider opening a new issue or post here the details which then show that it looks like the same as above.

If its the same as above, please consider providing more data.
